### PR TITLE
Add na_filter documentation to read_excel

### LINKED ISSUE #25453: read_excel na_filter undocumented?
#### Problem description

Empirically, `pandas.read_excel(foo, na_filter=False)` works as expected, but the na_filter argument doesn't appear to be mentioned in the read_excel docs.

### PR DESCRIPTION
**Pull request by bharatr21**
- [x] closes #25453
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry (Not needed)

### CODE REVIEW

**Comment by Bharat123rox** [comment]
CI test failure seems unrelated:
```
Exception: Cython-generated file 'pandas\_libs/algos.c' not found.
                    Cython is required to compile pandas from a development branch.
                    Please install Cython or download a release package of pandas.
```

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -134,8 +134,24 @@
     as NaN: '""" + fill("', '".join(
     sorted(_NA_VALUES)), 70, subsequent_indent="    ") + """'.
 keep_default_na : bool, default True
-    If na_values are specified and keep_default_na is False the default NaN
-    values are overridden, otherwise they're appended to.
+    Whether or not to include the default NaN values when parsing the data.
```
Comment (jreback) [inline pandas/io/excel/_base.py]: this is handled exactly the same as in pandas/io/parsers.py (e.g. read_csv); don't mind expand this sligthly, but these must be *the* same (also not that this is referenced in io.rst as well).

**Review by jreback — CHANGES_REQUESTED** [review]
pls merge master as well

File: pandas/io/excel/_base.py (hunk 1 above)
Comment (Bharat123rox) [reply pandas/io/excel/_base.py]: @jreback I'm not able to understand, isn't this the _exact  same_ docstring as that of `keep_default_na` in [`io/parsers.py`](https://github.com/pandas-dev/pandas/blob/b48d1ff87237cb68792590f4377e3be5c73d5ea7/pandas/io/parsers.py#L172-L186)?
If not, should I add the above two lines along with this?

**Comment by jreback** [comment]
closing as stale, ping if you'd like to continue.

**Pull request closed without merging**